### PR TITLE
Non local returns

### PR DESCRIPTION
**Pull request by francescoo22**

### CODE REVIEW

File: plugins/formal-verification/testData/diagnostics/good_contracts/inlining_captured.fir.diag.txt [hunk 1]
```diff
@@ -113,3 +113,39 @@ method global$fun_nested_call$fun_take$fun_take$T_Int$return$T_Int$fun_take$T_In
...
+      special$invoke_function_object(local$g)
+      anonymous$0 := 1
+      goto label$ret$1
+      special$invoke_function_object(local$g)
+      anonymous$1 := anonymous$2
+      goto label$ret$2
+      label label$ret$2
+    }
+    anonymous$0 := anonymous$1
+    goto label$ret$1
```
Comment (jesyspa) [inline plugins/formal-verification/testData/diagnostics/good_contracts/inlining_captured.fir.diag.txt:145]: Is this correct?  My impression was that `return@invoke` returns the value *to* `invoke`, but here it seems like it's returning it *from* `invoke`.

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/MethodConversionContext.kt [hunk 2]
```diff
@@ -19,12 +19,17 @@ interface MethodConversionContext : ProgramConversionContext {
     val signature: FunctionSignature
     val resolvedReturnVarName: MangledName
     val resolvedReturnLabelName: ReturnLabelName
+    val vanillaName: String?
```
[the remark this answers is not in the document]
Comment (francescoo22) [reply plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/MethodConversionContext.kt:22]: yes, it's the name of the function in the kotlin code

File: plugins/formal-verification/testData/diagnostics/good_contracts/inlining_captured.fir.diag.txt (hunk 1 above)
Comment (francescoo22) [reply plugins/formal-verification/testData/diagnostics/good_contracts/inlining_captured.fir.diag.txt:145]: yes, I misunderstood how `return@invoke` works in kotlin

File: plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/MethodConversionContext.kt [hunk 3]
```diff
@@ -19,12 +19,17 @@ interface MethodConversionContext : ProgramConversionContext {
...
+    val returnPointResolver: MutableMap<String, Pair<MangledName, ReturnLabelName>>
```
[the remark this answers is not in the document]
Comment (francescoo22) [reply plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/MethodConversionContext.kt]: Now I moved it in `ParameterResolver`, making it an implementation detail will require to add 4 methods to the interface

**Review by jesyspa — COMMENTED** [review]
It looks like the need for `isLambda` stems from the fact that you want a single `ParameterResolver` to be able to resolve *all* return values.  But this goes somewhat against our approach: we'd prefer this kind of chaining to go via the method conversion context, since that's what naturally gets a hierarchical structure for lambdas.  I feel like it should be possible to avoid the `isLambda` parameter and simplify the code, but maybe I'm missing something?

**Review by jesyspa — APPROVED** [review]
Some nits, but overall this looks good!

**Pull request merged**